factory/bootstrap/shardingFactory_test.go

```diff
@@ -0,0 +1,661 @@
+		t.Parallel()
+
+		nodesC, err := CreateNodesCoordinator(
+			&testscommon.ShuffleOutCloserStub{},
+			&testscommon.NodesSetupStub{},
+			config.PreferencesConfig{
+				DestinationShardAsObserver: "0",
+			},
+			&mock.EpochStartNotifierStub{},
+			&cryptoMocks.PublicKeyStub{
+				ToByteArrayStub: func() ([]byte, error) {
+					return nil, expectedErr
+				},
+			},
+			&testscommon.MarshalizerStub{},
+			&testscommon.HasherStub{},
+			&testscommon.RaterMock{},
+			&storage.StorerStub{},
+			&shardingMocks.NodeShufflerMock{},
+			0,
+			&bootstrapMocks.BootstrapParamsHandlerMock{},
+			0,
+			make(chan endProcess.ArgEndProcess, 1),
+			&nodeTypeProviderMock.NodeTypeProviderStub{},
+			&testscommon.EnableEpochsHandlerStub{},
+			&validatorInfoCacherMocks.ValidatorInfoCacherStub{},
+		)
+		require.True(t, errors.Is(err, expectedErr))
+		require.True(t, check.IfNil(nodesC))
+	})
+	t.Run("NewShuffledOutTrigger fails should error", func(t *testing.T) {
+		t.Parallel()
+
+		nodesC, err := CreateNodesCoordinator(
+			&testscommon.ShuffleOutCloserStub{},
+			&testscommon.NodesSetupStub{},
+			config.PreferencesConfig{
+				DestinationShardAsObserver: "0",
+			},
+			&mock.EpochStartNotifierStub{},
+			&cryptoMocks.PublicKeyStub{
+				ToByteArrayStub: func() ([]byte, error) {
+					return nil, nil // no error but nil pub key to force NewShuffledOutTrigger to fail
+				},
+			},
+			&testscommon.MarshalizerStub{},
+			&testscommon.HasherStub{},
+			&testscommon.RaterMock{},
+			&storage.StorerStub{},
+			&shardingMocks.NodeShufflerMock{},
+			0,
+			&bootstrapMocks.BootstrapParamsHandlerMock{},
+			0,
+			make(chan endProcess.ArgEndProcess, 1),
+			&nodeTypeProviderMock.NodeTypeProviderStub{},
+			&testscommon.EnableEpochsHandlerStub{},
+			&validatorInfoCacherMocks.ValidatorInfoCacherStub{},
+		)
+		require.NotNil(t, err)
+		require.True(t, check.IfNil(nodesC))
+	})
+	t.Run("NewIndexHashedNodesCoordinator fails should error", func(t *testing.T) {
+		t.Parallel()
+
+		nodesC, err := CreateNodesCoordinator(
+			&testscommon.ShuffleOutCloserStub{},
+			&testscommon.NodesSetupStub{},
+			config.PreferencesConfig{
+				DestinationShardAsObserver: "0",
+			},
+			&mock.EpochStartNotifierStub{},
+			&cryptoMocks.PublicKeyStub{
+				ToByteArrayStub: func() ([]byte, error) {
+					return []byte("public key"), nil
+				},
+			},
+			nil, // force NewIndexHashedNodesCoordinator to fail
+			&testscommon.HasherStub{},
+			&testscommon.RaterMock{},
+			&storage.StorerStub{},
+			&shardingMocks.NodeShufflerMock{},
+			0,
+			&bootstrapMocks.BootstrapParamsHandlerMock{},
+			0,
+			make(chan endProcess.ArgEndProcess, 1),
+			&nodeTypeProviderMock.NodeTypeProviderStub{},
+			&testscommon.EnableEpochsHandlerStub{},
+			&validatorInfoCacherMocks.ValidatorInfoCacherStub{},
+		)
+		require.NotNil(t, err)
+		require.True(t, check.IfNil(nodesC))
+	})
+	t.Run("NewIndexHashedNodesCoordinator fails should error", func(t *testing.T) {
+		t.Parallel()
```

Comment:
this test name and the previous one have the same name, rename here?

Reply:
fixed